encrypted-media/scripts/reset-src-after-setmediakeys.js

```diff
@@ -0,0 +1,68 @@
+function runTest(config) {
+    async_test(function (test) {
+        var mediaKeys;
+        var mediaSource;
+        var encryptedEventIndex = 0;
+        var video = config.video;
+        var keysystem = config.keysystem;
+        var configuration = {   initDataTypes: [ config.initDataType ],
+            audioCapabilities: [ { contentType: config.audioType } ],
+            videoCapabilities: [ { contentType: config.videoType } ],
+            sessionTypes: [ 'temporary' ] };
+
+        assert_not_equals(video, null);
+
+        // Alternate content to be played. These files must be the same format.
+        var alternateContent = config.alternateContent;
+
+        var onEncrypted = function (event) {
+            ++encryptedEventIndex;
+            assert_not_equals(video.mediaKeys, null);
+            assert_true(video.mediaKeys === mediaKeys);
+
+            // This event is fired once for the audio stream and once
+            // for the video stream each time .src is set.
+            if (encryptedEventIndex == 2) {
+                // Finished first video; set src to a different video.
+                config.videoPath = alternateContent.video.path;
+                //TODO set different audio path, audio content needed
+                return testmediasource(config)
```

Comment:
Since we are using MediaSource, we could actually use the same content again, since we will create a second MediaSource and there is no way the browser could know it is the same data being provided as was provided to the first MediaSource (we are even fetching it again).